src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_endpoint.py

```diff
@@ -64,6 +64,41 @@ def remove_akamai_access_control(client, account_name, resource_group_name, stre
     return client.update(resource_group_name, account_name, streaming_endpoint_name, streaming_endpoint)
 
 
+def update_streaming_endpoint(client, resource_group_name, account_name, streaming_endpoint_name,  # pylint: disable=too-many-locals
+                              tags=None, cross_domain_policy=None, ips=None, client_access_policy=None,
+                              description=None, max_cache_age=None,
+                              cdn_provider=None, cdn_profile=None, custom_host_names=None):
+    from azure.mgmt.media.models import (IPAccessControl, StreamingEndpointAccessControl)
+
+    streaming_endpoint = client.get(resource_group_name, account_name, streaming_endpoint_name)
+
+    allow_list = []
+    if ips is not None:
+        for ip in ips:
+            allow_list.append(create_ip_range(streaming_endpoint_name, ip))
+
+    streaming_endpoint_access_control = None
+    if ips is not None:
+        streaming_endpoint_access_control = StreamingEndpointAccessControl(ip=IPAccessControl(allow=allow_list))
+
+    policies = create_cross_site_access_policies(client_access_policy, cross_domain_policy)
+
+    cdn_enabled = cdn_profile is not None or cdn_provider is not None
+
+    streaming_endpoint.max_cache_age = max_cache_age
+    streaming_endpoint.tags = tags
+    streaming_endpoint.description = description
+    streaming_endpoint.custom_host_names = custom_host_names
+    streaming_endpoint.cdn_profile = cdn_profile
+    streaming_endpoint.cdn_enabled = cdn_enabled
+    streaming_endpoint.cdn_provider = cdn_provider
+    streaming_endpoint.cross_site_access_policies = policies
+    streaming_endpoint.access_control = streaming_endpoint_access_control
+
+    return client.update(resource_group_name=resource_group_name, account_name=account_name,
```

Comment:
Make sure to avoid updating a property if it isn't set by the user in the CLI... This avoids setting None to an existing property you don't want to update.